core/modules/field/field.crud.inc

```diff
@@ -261,15 +261,6 @@ function field_validate_field($field, $update = FALSE) {
       throw new FieldException(t('The field "@name" could not be created because a database table with the name "@table" already exists.', array('@name' => $field['field_name'], '@table' => 'field_revision_' . $field['field_name'])));
     }
 
-    // Ensure the field name is unique.
-    $prior_field = field_read_field($field['field_name'], array('include_inactive' => TRUE, 'include_deleted' => TRUE));
-    if (!empty($prior_field)) {
-      $message = $prior_field['active']?
-        t('The field "@name" already exists.', array('@name' => $field['field_name'])):
-        t('The field "@name" already exists, but it is inactive. To create a field with the same name, the configuration and data from the old field must be manually deleted.', array('@name' => $field['field_name']));
```

Comment:
I don't think we should be entirely removing these checks. Is there a way we can keep them while still having the config import on install work properly?

Reply:
Since this "Validates a field configuration," would it be reasonable to place a `$config->isNew()` check (and related code) at an appropriate place in the function?